### PR TITLE
Align iOS and Android methods

### PR DESCRIPTION
**Pull request by igstewart3**
Ensures that iOS and Android are using the same method names. Also added 
some missing methods to both sides and updated any methods using deprecated native methods.

### CODE REVIEW

File: android/src/main/java/com/reactlibrary/RNCarnivalModule.java [hunk 1]
```diff
@@ -423,15 +433,21 @@ public void onFailure(Error error) {
     }
 
     @ReactMethod
-    public void presentMessageDetail(String messageId) {
+    public void presentMessageDetail(ReadableMap message) {
```
[the remark this answers is not in the document]
Comment (josecoelho) [reply android/src/main/java/com/reactlibrary/RNCarnivalModule.java]: Oops. I see it now. All internal code from iOS requires the Message object. :/

Leave it as is...

👍

File: example/app.js [hunk 2]
```diff
@@ -73,11 +73,17 @@ export default class App extends Component<{}> {
     Carnival.setUserId("person");
     Carnival.setUserEmail("person@domain.com");
 
-    //Carnival.setGeoIPTrackingEnabled(true);
+    Carnival.setGeoIPTrackingEnabled(true);
 
-    //Carnival.setCrashHandlersEnabled(true);
+    // Carnival.setCrashHandlersEnabled(true);
```
[the remark this answers is not in the document]
Comment (igstewart3) [reply example/app.js]: I'm not sure why they were added originally, but I would guess it was just so there was an example of how all the available methods could/should be called in the example app. Probably a good call to move them out to the docs though.

File: android/src/main/java/com/reactlibrary/RNCarnivalModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (igstewart3) [reply android/src/main/java/com/reactlibrary/RNCarnivalModule.java]: Yeah, would have gone for strings if I could have.

File: example/index.android.js [hunk 3]
```diff
@@ -70,11 +70,17 @@ class ReactNativeSampleApp extends Component {
 
     Carnival.setUserId("person@domain.com");
 
-    //Carnival.setGeoIPTrackingEnabled(true);
+    Carnival.setGeoIPTrackingEnabled(true);
 
     //Carnival.setCrashHandlersEnabled(true);
```
[the remark this answers is not in the document]
Comment (igstewart3) [reply example/index.android.js]: I'll remove them now and add a task to update the RN docs.

**Comment by igstewart3** [comment]
Updated to remove unused methods from example app. Will be added to documentation instead.

**Comment by igstewart3** [comment]
Final change: updated to new Android SDK release and fixed some tests.

**Pull request merged**